tweetnacl-multiplatform/vendor/tweetnacl/tweetnacl.c

```diff
@@ -807,9 +807,21 @@ int crypto_sign_open(u8 *m,u64 *mlen,const u8 *sm,u64 n,const u8 *pk)
 
 // curve additions
 
-int is_on_curve(const u8 p[32]) {
+int is_on_curve(const u8 p[32])
+{
   gf q[4];
-  return unpackneg(q,p);
+
+  return unpackneg(q, p);
+}
+
+int public_key_from_secret(u8 *sk, u8 *pk)
```

Comment:
key point here: this is these two steps from `crypto_sign_keys`

https://github.com/avianlabs/solana-kotlin/pull/142/files#diff-92529ce799129d8771b476f2d8ec0a4c43c4908201bb209f636a747b87622811R665-R666

which is what derives the public key. these primitives are not exported by tweetnacl so we need to write our own C function like with `is_on_curve`